### ISSUE TITLE
Silent Install option to disable Runtime check?

### ISSUE THREAD

**Issue by OfficialEsco** [Sigil-Ebook/Sigil#634]
Hey. i believe this is not the preferred issue reporting location but i cba going to the forum as i am not a user of the software.

I am trying to add Sigil version 1.7.0 to the WinGet repo https://github.com/microsoft/winget-pkgs/pull/23907, however there is a new popup prompt in this version which i cannot find the switch to bypass
![image](https://user-images.githubusercontent.com/15158490/128477520-2d1605a7-67f7-4f6b-a5d8-0ce83f390109.png)

So i wonder, is there any internal switch to disable this popup?

**Comment by dougmassay** [comment]
That popup is not new to Sigil v1.7.0. It's been around for several releass now. But no, there is currently no internal option to silence it.

**Comment by OfficialEsco** [comment]
When retesting 1.6.0 you seem to be correct..
However it passed both Manual and Automatic Validation before so something is different..

In fact, when going trough the installer manually i found this weirdness

Default installation mode is full: 
![image](https://user-images.githubusercontent.com/15158490/128493094-9fa1b953-9e31-4f9c-8bcb-66f067332359.png)
When you click Full installation:
![image](https://user-images.githubusercontent.com/15158490/128493127-8cf09d02-4608-4c9d-9e07-83093ffd2f82.png)

**Comment by dougmassay** [comment]
Sorry, but I'm not quite certain if there's still a question for me or not. The installer hasn't been changed at all since between the 1.4.3 and the 1.5.0/1 versions of Sigil. I'm afraid I don't take silent installs into consideration when I create the Sigil Windows installers.

**Comment by OfficialEsco** [comment]
Kinda left a question without a question mark once i found the issue which is why its not closed 😬
Why does `/CURRENTUSER /COMPONENTS="vcruntimeuser"` not work? Or what is it supposed to do?
Because i assume this is not the right component `/CURRNETUSER /COMPONENTS="vcruntimeadmin"` (which also does not work)

**Comment by dougmassay** [comment]
I honestly have no idea what the stock InnoSetup command line switches are supposed to do. I don't use them.

I will look into changing the installer to better accommodate silent installs. Now that the runtimes for different versions of Visual Studio are consolidated (and backward compatible starting with VS2015), it may be time to consider removing the runtime installer component from Sigil's installer altogether.

I'll leave this issue open while I work on it.

**Comment by dougmassay** [comment]
OK, this should be taken care of in the next Sigil release. I enabled the use of the /ALLUSERS and /CURRENTUSER on the command line. This gets rid of the dialog for that choice. I also changed to using a SuppressableMsgBox that can be suppressed on the command line with the /SUPPRESSMSGBOXES option (which must be used in conjunction with the /SILENT or /VERYSILENT options).

The VC++ runtime check/installation is enabled by default for both /ALLUSERS and /CURRENTUSER, so you'll need to override the /COMPONENTS option if you want to skip it. To install the desktop icon and associated .epub files with Sigil something like: /COMPONENTS="dicon,*afiles" should work. Leave off the "*afiles" if you don't want to associate .epub file with Sigil. Leave off "dicon" if you don't want to install a desktop icon for Sigil.  /COMPONENTS="" would install only the Sigil files and nothing else (with the exception of the start menu icons that can be disabled with the /NOICONS option).

A completely unattended Sigil install for the current user with no VC++ check (with desktop icon and .epub file associations) would look something like:

`.\Sigil-1.8.0-Setup.exe /CURRENTUSER /NORESTART /VERYSILENT /SUPPRESSMSGBOXES /COMPONENTS="dicon,*afiles"`

All users with a vc++ check would be:

`.\Sigil-1.8.0-Setup.exe /ALLUSERS /NORESTART /VERYSILENT /SUPPRESSMSGBOXES /COMPONENTS="dicon,*afiles,vcruntimeadmin"`

The following would skip the desktop icon, the start menu entries, the .epub file association, and the vc++ runtime check (for the current user):

`.\Sigil-1.7.0-Setup.exe /CURRENTUSER /NORESTART /VERYSILENT /SUPPRESSMSGBOXES /NOICONS /COMPONENTS=""`

Specifying no /COMPONENTS would get you the default file install, the desktop icon install, and the vc++ check/install. But with /SILENT (or /VERYSILENT) and /SUPPRESSMSGBOXES supplied, there would be no prompt to content with.
